queries/latex/highlights.scm

```diff
@@ -0,0 +1,165 @@
+;; Math
+[
+ (displayed_equation)
+ (inline_formula)
+] @text.math
+
+;; This highlights the whole environment like vimtex does
+((environment
+  (begin
+   name: (word) @_env)) @text.math
+   (#match? @_env "^(displaymath|equation|multline|eqnarray|align)[*]?$"))
+
+;; This at the begining of the file would be the alternative to highlight
+;; only the interior of the environment
+;((environment
+  ;(begin
+   ;name: (word) @_env) @none
+   ;(end) @none) @text.math
+   ;(#match? @_env "^(displaymath|eqn|eqnarray)[*]?$"))
+
+[
+  (generic_command_name)
+  "\\newcommand"
+  "\\renewcommand"
+  "\\DeclareRobustCommand"
+  "\\DeclareMathOperator"
+  "\\newglossaryentry"
+  "\\caption"
+  "\\cite"
+  "\\label"
+  "\\newlabel"
+  "\\label"
+  "\\ref"
+] @function.macro
+
+(comment) @comment
+
+(bracket_group) @parameter
+
+[(math_operator) "="] @operator
+
+[
+  "\\usepackage"
+  "\\documentclass"
+  "\\input"
+  "\\include"
+  "\\subfile"
+  "\\subfileinclude"
+  "\\subfileinclude"
+  "\\includegraphics"
+  "\\addbibresource"
+  "\\bibliography"
+  "\\includesvg"
+  "\\includeinkscape"
+  "\\usepgflibrary"
+  "\\usetikzlibrary"
+] @include
+
+[
+  "\\part"
+  "\\chapter"
+  "\\section"
+  "\\subsection"
+  "\\subsubsection"
+  "\\paragraph"
+  "\\subparagraph"
+] @type
+
+"\\item" @punctuation.special
+
+((word) @punctuation.delimiter
+(#eq? @punctuation.delimiter "&"))
+
+["$" "\\[" "\\]" "\\(" "\\)"] @punctuation.delimiter
+
+(label_definition
+ name: (_) @text.reference)
+(label_reference
+ label: (_) @text.reference)
+(equation_label_reference
+ label: (_) @text.reference)
+(label_reference
+ label: (_) @text.reference)
+(label_number
+ label: (_) @text.reference)
+
+(citation
+ key: (word) @text.reference)
+
+(key_val_pair
+  key: (_) @method ; like JS
+  value: (_))
+
+["(" ")" "[" "]" "{" "}"] @punctuation.bracket
```

Comment:
This may be getting a bit nit-picky, but I don't think that's a good idea since it will also capture `(...)` in text mode, where it doesn't have any special meaning. Unless you want to only capture this in math mode (which I wouldn't suggest for this PR), I would
1. only capture `[]`/`{}` as `@puncuation.delimiter` (since these are actually syntax elements of TeX)
2. capture `$` and `\[` etc. as as `@text.math` (same as `\begin{equation}` etc.)

Only question is how this would apply to `\{` etc. (which _are_ brackets)?

Reply:
> capture $ and \[ etc. as as @text.math (same as \begin{equation} etc.)

but didn't you just right now say to add them? The math environment is highlighted as whole anyways

Reply:
Seems to actually work well. I would just remove the `()` here; the naming suggestion is not crucial and up to you.

Reply:
> Only question is how this would apply to \{ etc. (which are brackets)?

Reply:
> > capture $ and \[ etc. as as @text.math (same as \begin{equation} etc.)
> 
> but didn't you just right now say to add them? The math environment is highlighted as whole anyways

I did, yes, and I apologize for the less-than-well structured review... 😞 

To be clear: Here I am only talking about the name of the capture group (arguing that `$` should behave the same as `\begin{equation}`). The fact _that_ they are captured is good as is!